researchstudio-sat/webofneeds
# Bump javascript dependencies

## Fixes #2493 

This bumps all of our dependencies to the latest version, among them [`rdf-canonize`](https://github.com/digitalbazaar/rdf-canonize). The new version of this library does not need any native code, which should make building on Windows a much smoother experience.

Among other things, [`prettier`](https://github.com/prettier/prettier) is also updated. This means some styling rules have changed, hence the big changelog.

> **Warning**
> This PR touches a lot of files since the styling rules have changed. This *will* cause conflicts with existing PRs

## Testing

1. Run `npm install`
2. Run `npm run watch`
3. Check that everything compiles correctly
4. Open the page in the browser and make sure it still functions the same. Errors in this PR should make huge parts of the application non-functional, so if everything seems fine it probably is.

> **Warning**
> When this is merged everyone will need to run `npm install` on their machine, and if they are using a `prettier` extension in their IDE they might need to restart said IDE (this is the case for VSCode at least)
